src/main/java/com/dungeoncrawler/Javiarenka/shop/Shop.java

```diff
@@ -12,38 +11,23 @@
 
 public class Shop {
 
-    private static List<Armor> allAvailableArmor = new ArrayList<>(Arrays.asList(null, null, null, null, null)); // todo
-    private static List<Weapon> allAvailableWeapon = new ArrayList<>(Arrays.asList(null, null, null, null, null)); // todo
+    private static List<Equipment> allAvailableEquipment = new ArrayList<>(Arrays.asList(null, null, null, null, null)); // todo: list of available Equipment
 
-    public static List<Armor> getNewArmor(Object object) {
+    public static List<Equipment> getNewEquipment(Object object) {
```

Comment:
Dlaczego właściwie ta metoda przyjmuje dowolny obiekt, a nie po prostu obiekt klasy Hero? Przecież tylko Herosi będą u nas mogli kupować ekwipunek.

Reply:
metoda getNewEquipment() przyjmuje obiekt klasy Object, dlatego, że w środku ma zagnieżdżoną metodę validation(), który przyjmuje obiekt i w zależności czy jest to obiekt klasy Hero czy innej odpalają się odpowiednie linijki kodu - jeżeli ta metoda przyjmowałaby Hero wprost, kod nie kompilowałby się właśnie przez metodę validation() - mogę jeszcze pokombinować ;)

Reply:
Jasne, nie jest to zły pomysł, chociaż moim zdaniem bardziej użytwczny jeśli getNewEquipment mogłaby przyjmować jakieś różne obiekty nie dziedziczące po sobie, jak to czasem dzieje się we wzorcu projektowym fabryki (Factory). Tu jednak przedmioty będzie kupował zawsze heros, a nam lepiej będzie jak Java krzyknie nam "nie ma takiej metody jak getNewEquipment(Smerf smerf), niż jak coś wywróci nam się w kompilacji i będziemy musieli szukać tego w StackTrace :)

Reply:
Ale PRa i tak approve-nąłem, bo moim zdaniem nie jest to mega błąd i działa. Co więcej widzę, że czasem takie podejście może być nawet lepsze, tylko wydaje mi się, że akurat nie w tym przypadku.

Reply:
sprawdziłam i bez walidacji z trzech metod() zostałaby tak naprawdę jedna, przyjmująca obiekt klasy Hero XD mogłabym póki co za-komentować (so unprofessional) te dwie metody na ewentualną przyszłość, a póki co metoda() działaby tak jak miała od początku czyli z obiektem klasy Hero jako parametr